apps/etl/tasks.py

```diff
@@ -0,0 +1,273 @@
+        "validation_error": validation_error,
+    }
+    return validation_data
+
+
+def validate_gdacs_geometry_data(resp_data, hazard_type=None):
+    try:
+        resp_data_for_validation = json.loads(resp_data.decode("utf-8"))
+        GdacsEventsGeometryData(**resp_data_for_validation)
+        validation_error = ""
+    except ValidationError as e:
+        validation_error = e.json()
+    validation_data = {
+        "status": ExtractionData.ValidationStatus.FAILED if validation_error else ExtractionData.ValidationStatus.SUCCESS,
+        "validation_error": validation_error if validation_error else "",
+    }
+    return validation_data
+
+
+def manage_duplicate_file_content(source, hash_content, instance, response_data, file_name):
+    duplicate_file_content = ExtractionData.objects.filter(source=source, file_hash=hash_content)
+    if duplicate_file_content:
+        instance.resp_data = duplicate_file_content.first().resp_data
+        instance.revision_id = duplicate_file_content.first()
+    else:
+        instance.resp_data.save(file_name, ContentFile(response_data))
+    instance.save()
+
+
+def hash_file_content(content):
+    """
+    Compute the hash of a file using the specified algorithm.
+    :return: Hexadecimal hash of the file
+    """
+    file_hash = hashlib.sha256(content).hexdigest()
+    return file_hash
+
+
+def store_extraction_data(response, validate_source_func, parent_id=None, instance_id=None, hazard_type=None):
+    file_extension = response.pop("file_extension")
+    file_name = f"gdacs.{file_extension}"
+    resp_data = response.pop("resp_data")
+    if not instance_id:
+        gdacs_instance = ExtractionData.objects.create(**response, parent_id=parent_id)
+    else:
+        gdacs_instance = ExtractionData.objects.get(id=instance_id)
+        for key, value in response.items():
+            setattr(gdacs_instance, key, value)
+        gdacs_instance.save()
+
+    if resp_data and not response["resp_code"] == 204:
+        resp_data_content = resp_data.content
+        # Source validation
+        if validate_source_func:
+            gdacs_instance.source_validation_status = validate_source_func(resp_data_content, hazard_type)["status"]
+            gdacs_instance.content_validation = validate_source_func(resp_data_content, hazard_type)["validation_error"]
+
+        # duplicate file content check
+        hash_content = hash_file_content(resp_data_content)
+        manage_duplicate_file_content(
+            source=ExtractionData.Source.GDACS,
+            hash_content=hash_content,
+            instance=gdacs_instance,
+            response_data=resp_data_content,
+            file_name=file_name,
+        )
+    return gdacs_instance
+
+
+@shared_task(bind=True, max_retries=2, default_retry_delay=5)
+def scrape_population_exposure_data(self, parent_id, event_id: int, hazard_type: str, parent_transform_id):
+    url = f"https://www.gdacs.org/report.aspx?eventid={event_id}&eventtype={hazard_type}"
```

Comment:
Have you looked at the current GDACs ETL in R? https://github.com/IFRCGo/GCDB/blob/main/RCode/MainlyHazardData/GetGDACS.R — I think it's really important to conceptually follow the data extraction plan in the R code. We don't necessarily want to add new attributes or logic. cc @emmanuelmathot @subinasr @Rup-Narayan-Rajbanshi @ranjan-stha

Reply:
And to support @geohacker's commen, please check the interface of the GDACS Transformer in [pystac-monty ](https://github.com/IFRCGo/pystac-monty) repository, especially the [tests](https://github.com/IFRCGo/pystac-monty/blob/main/tests/extensions/test_gdacs.py) that exemplify how to call the transformer for a given event with several data sources (event_data, geometry, impacts...). Basically, the transformer ingest the raw data source (+origin URL and data tag) and outputs the items ready to be loaded then in the STAC API